tensorrt_llm/_torch/modules/linear.py

```
@@ -1457,22 +1525,54 @@ def load_weights_vanilla(self, module: Linear, weights: List[Dict]) -> None:
 
-        input_scale, weight_scales, weight_scale_2, alpha = self.load_weight_scales(
-            weights,
-            tp_size=module.tp_size,
-            tp_rank=module.tp_rank,
-            tp_mode=module.tp_mode)
-        weight_scale = torch.cat(weight_scales, 0)
-        weight_scale = torch.ops.trtllm.block_scale_interleave(weight_scale)
+        weight_mode = module.weights_loading_config.weight_mode
 
+        for shard_key, weight in zip(('q', 'k', 'v'),
+                                     (q_weight, k_weight, v_weight)):
```

Ruff (0.15.10)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)